eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Security/PolicyProvider/PoliciesConfigBuilder.php

```diff
@@ -14,28 +14,47 @@ class PoliciesConfigBuilder extends ContainerConfigBuilder
 {
     public function addConfig(array $config)
     {
-        $policyMap = [];
         $previousPolicyMap = [];
 
+        if ($this->containerBuilder->hasParameter('ezpublish.api.role.policy_map')) {
+            $previousPolicyMap = $this->containerBuilder->getParameter('ezpublish.api.role.policy_map');
+        }
+
         // We receive limitations as values, but we want them as keys to be used by isset().
         foreach ($config as $module => $functionArray) {
             foreach ($functionArray as $function => $limitationCollection) {
-                $policyMap[$module][$function] = array_fill_keys((array)$limitationCollection, true);
-            }
-        }
+                if (null !== $limitationCollection && $this->policyExist($previousPolicyMap, $module, $function)) {
+                    $limitations = array_merge_recursive($previousPolicyMap[$module][$function], array_fill_keys((array)$limitationCollection, true));
+                } else {
+                    $limitations = array_fill_keys((array)$limitationCollection, true);
+                }
 
-        if ($this->containerBuilder->hasParameter('ezpublish.api.role.policy_map')) {
-            $previousPolicyMap = $this->containerBuilder->getParameter('ezpublish.api.role.policy_map');
+                $previousPolicyMap[$module][$function] = $limitations;
+            }
         }
 
         $this->containerBuilder->setParameter(
             'ezpublish.api.role.policy_map',
-            array_merge_recursive($previousPolicyMap, $policyMap)
+            $previousPolicyMap
         );
     }
 
     public function addResource(ResourceInterface $resource)
     {
         $this->containerBuilder->addResource($resource);
     }
+
+    /**
+     * Checks if policy for module and function exist in Policy Map.
+     *
+     * @param array $policyMap
+     * @param string $module
+     * @param string $function
+     *
+     * @return bool
+     */
+    private function policyExist(array $policyMap, string $module, string $function)
```

Comment:
1. Typo: `policyExists`
2. Don't use PHP 7.x type hinting for kernel `6.x` (keep here `array` only and of course keep hinting in PhpDoc).